code/game/gamemodes/objectives/open/explosion.dm

```diff
@@ -0,0 +1,88 @@
+/datum/objective/open/explosion
+	name = "detonate explosive"
+	explanation_text = "Obtain and detonate an explosive device within %DEPARTMENT%."
+	weight = 6
+	var/selected_area
+	var/list/valid_areas = list(
+		"medical" = list(/area/medical),
+		"engineering" = list(/area/engineering, /area/engine),
+		"security" = list(/area/security),
+		"the cargo bay" = list(/area/quartermaster, /area/cargo),
+		"the bridge" = list(/area/bridge),
+		"the communications relay" = list(/area/comms, /area/server),
+		"the science lab" = list(/area/science),
+		"the research division server room" = list(/area/science/server),
+		// Anywhere monitored by the AI will do
+		"the AI's facilities" = list(/area/aisat, /area/ai_monitored),
+		"the Captain's office" = list(/area/crew_quarters/heads/captain),
+		"the Head of Personnel's office" = list(/area/crew_quarters/heads/hop)
+	)
+	var/success = FALSE
+	var/devistation = 0
+	var/heavy = 0
+	var/light = 0
+
+/datum/objective/open/explosion/New(text)
+	. = ..()
+	//Register for the signals
+	RegisterSignal(SSdcs, COMSIG_GLOB_EXPLOSION, PROC_REF(on_explosion))
+
+/datum/objective/open/explosion/Destroy(force, ...)
+	UnregisterSignal(SSdcs, COMSIG_GLOB_EXPLOSION)
+	. = ..()
+
+/datum/objective/open/explosion/proc/on_explosion(datum/source, turf/epicenter, devastation_range, heavy_impact_range, light_impact_range, took, orig_dev_range, orig_heavy_range, orig_light_range)
+	if(!light_impact_range && !heavy_impact_range && !devastation_range)
+		return
+	// Get all the areas that we effect
```

Comment:
this could check the source of the explosion, and if it was deployed / fingerprinted by the person doing it.
Not required but something to consider

Reply:
fingerprint isn't reliable because a case that you brainwash & persuade a person to do it won't be counted.

Reply:
I suppose

Reply:
The point of these objectives is to be tracked, but weakly tracked. False negatives are much worse than false positives, so I'm keeping them open